x-pack/plugins/apm/public/components/shared/environment_filter/index.tsx

```diff
@@ -5,106 +5,37 @@
  * 2.0.
  */
 
-import { EuiSelect } from '@elastic/eui';
-import { i18n } from '@kbn/i18n';
 import { History } from 'history';
 import React from 'react';
 import { useHistory, useLocation } from 'react-router-dom';
-import {
-  ENVIRONMENT_ALL_SELECT_OPTION,
-  ENVIRONMENT_NOT_DEFINED,
-} from '../../../../common/environment_filter_values';
 import { fromQuery, toQuery } from '../links/url_helpers';
-import { FETCH_STATUS } from '../../../hooks/use_fetcher';
-import { Environment } from '../../../../common/environment_rt';
-import { useEnvironmentsContext } from '../../../context/environments_context/use_environments_context';
+import { EnvironmentSelect } from '../environment_select';
 
 function updateEnvironmentUrl(
   history: History,
   location: ReturnType<typeof useLocation>,
-  environment: string
+  environment?: string
 ) {
   const nextEnvironmentQueryParam = environment;
   history.push({
     ...location,
     search: fromQuery({
       ...toQuery(location.search),
-      environment: nextEnvironmentQueryParam,
+      ...(environment &&
+        environment.length > 0 && { environment: nextEnvironmentQueryParam }),
     }),
   });
 }
 
-const SEPARATOR_OPTION = {
-  text: `- ${i18n.translate(
-    'xpack.apm.filter.environment.selectEnvironmentLabel',
-    { defaultMessage: 'Select environment' }
-  )} -`,
-  disabled: true,
-};
-
-function getOptions(environments: string[]) {
-  const environmentOptions = environments
-    .filter((env) => env !== ENVIRONMENT_NOT_DEFINED.value)
-    .map((environment) => ({
-      value: environment,
-      text: environment,
-    }));
-
-  return [
-    ENVIRONMENT_ALL_SELECT_OPTION,
-    ...(environments.includes(ENVIRONMENT_NOT_DEFINED.value)
-      ? [ENVIRONMENT_NOT_DEFINED]
-      : []),
-    ...(environmentOptions.length > 0 ? [SEPARATOR_OPTION] : []),
-    ...environmentOptions,
-  ];
-}
-
 export function ApmEnvironmentFilter() {
-  const { status, environments, environment } = useEnvironmentsContext();
-
-  return (
-    <EnvironmentFilter
-      status={status}
-      environment={environment}
-      environments={environments}
-    />
-  );
-}
-
-export function EnvironmentFilter({
```

Comment:
is this not being used by the UX app?

Reply:
it seems they're using their own implementation 
https://github.com/elastic/kibana/blob/main/x-pack/plugins/ux/public/components/app/rum_dashboard/environment_filter/index.tsx

Reply:
Ok, maybe they copied+pasted it over there. Great that we can clean this up now 👍